install/swiftly-install.sh

```diff
@@ -71,9 +115,57 @@ bold () {
     echo "$(tput bold)$1$(tput sgr0)"
 }
 
+# Fetch the list of required system dependencies from the apple/swift-docker
+# repository and attempt to install them using the system's package manager.
+#
+# $docker_platform_name, $docker_platform_version, and $package manager need
+# to be set before calling this function.
+install_system_deps () {
+    if [[ "$(id --user)" != "0" ]] && ! has_command sudo ; then
+        echo "Warning: sudo not installed and current user is not root, skipping system dependency installation."
+        return
+    elif ! has_command "$package_manager" ; then
+        echo "Warning: package manager \"$package_manager\" not found, skipping system dependency instllation."
+        return
+    fi
+
+    dockerfile_url="https://raw.githubusercontent.com/apple/swift-docker/main/nightly-main/$docker_platform_name/$docker_platform_version/Dockerfile"
+    dockerfile="$(curl --silent --retry 3 --location --fail $dockerfile_url)"
+    if [[ "$?" -ne 0 ]]; then
+        echo "Error enumerating system dependencies, skipping installation of system dependencies."
+    fi
+
+    # Find the line number of the RUN command associated with installing system dependencies.
+    beg_line_num=$(printf "$dockerfile" | grep -n --max-count=1 "$package_manager.*install" | cut -d ":" -f1)
+
+    # Starting from there, find the first line that doesn't have the same level of indentation.
+    relative_end_line_num=$(printf "$dockerfile" |
+                                tail --lines=+"$((beg_line_num + 1))" |
+                                grep -n --max-count=1 --invert-match "^[ ]\{2,4\}[^& ]" | cut -d ":" -f1)
+    end_line_num=$((beg_line_num + relative_end_line_num - 1))
+
+    # Read the lines between those two, deleting any spaces and backslashes.
+    readarray -t package_list < <(printf "$dockerfile" | sed -n "$((beg_line_num + 1)),${end_line_num}p" | sed -r 's/[\ ]//g')
+
+    install_args=(--quiet)
+    if [[ "$DISABLE_CONFIRMATION" == "true" ]]; then
+        install_args+=(-y)
+    fi
+
+    # Disable errexit since failing to install system dependencies is not swiftly installation-fatal.
+    set +o errexit
+    if [[ "$(id --user)" == "0" ]]; then
+        "$package_manager" install "${install_args[@]}" "${package_list[@]}"
```

Comment:
Is there a chance on ubuntu that you need to call `apt-get update` first?

Reply:
Yeah, especially for a fresh docker image. That said, I think it would be better for users to do that outside of the script, since in most cases calling update shouldn't be needed, and we don't want to do it unnecessarily since it can involve downloading a decent amount of data.

Reply:
If a package install fails is it worthwhile pointing out to the user they might want to run `apt-get update`

Reply:
good idea, done